src/resty/aws/service/rds/signer.lua

```diff
@@ -0,0 +1,134 @@
+--   hostname = db_hostname,
+--   username = "db_user",
+--   port = db_port,
+--   region = nil,              -- will be inherited from `aws`
+--   credentials = nil,         -- will be inherited from `aws`
+-- }
+--
+-- -- use the 'signer' to generate the token, whilst overriding some options
+-- local auth_token, err = signer:getAuthToken {
+--   username = "another_user"  -- this overrides the earlier provided config above
+-- }
+--
+-- if err then
+--   ngx.log(ngx.ERR, "Failed to build auth token: ", err)
+--   return
+-- end
+--
+-- local pg = pgmoon.new({
+--   host = db_hostname,
+--   port = db_port,
+--   database = db_name,
+--   user = "another_user",
+--   password = auth_token,
+--   ssl = true,
+-- })
+--
+-- local flag, err = pg:connect()
+-- if err then
+--  ngx.log(ngx.ERR, "Failed to connect to database: ", err)
+--  return
+-- end
+--
+-- -- Test query
+-- assert(pg:query("select * from users where status = 'active' limit 20"))
+local function getAuthToken(self, opts) --endpoint, region, db_user)
+  opts = setmetatable(opts or {}, { __index = self.config }) -- lookup missing params in inherited config
+
+  local region = assert(opts.region, "parameter 'region' not set")
+  local hostname = assert(opts.hostname, "parameter 'hostname' not set")
+  local port = assert(opts.port, "parameter 'port' not set")
+  local username = assert(opts.username, "parameter 'username' not set")
+
+  local endpoint = hostname .. ":" .. port
+  if endpoint:sub(1,7) ~= "http://" and endpoint:sub(1,8) ~= "https://" then
+    endpoint = "https://" .. endpoint
+  end
+
+  local query_args = "Action=connect&DBUser=" .. username
+
+  local canonical_request_url = endpoint .. "/?" .. query_args
+  local scheme, host, port, path, query = unpack(httpc:parse_uri(canonical_request_url, false))
+  local req_data = {
+    method = "GET",
+    scheme = scheme,
+    tls = scheme == "https",
+    host = host,
+    port = port,
+    path = path,
+    query = query,
+    headers = {
+      ["Host"] = host .. ":" .. port,
+    },
+  }
+
+  local presigned_request, err = presign_awsv4_request(self, req_data, opts.signingName, region, RDS_IAM_AUTH_EXPIRE_TIME)
+  if err then
+    return nil, err
+  end
+
+  return presigned_request.host .. ":" .. presigned_request.port .. presigned_request.path .. "?" .. presigned_request.query
+end
+
+
+-- signature: intended to be a method on the RDS service object, rds_instance == self in that case
+return function(rds_instance, config)
+  local token_instance = {
+    config = {},
+    getAuthToken = getAuthToken,  -- injected method for token generation
+  }
+
+  -- first copy the inherited config elements NOTE: inherits from AWS, not the rds_instance!!!
+  for k,v in pairs(rds_instance.aws.config) do
+    token_instance.config[k] = v
+  end
+
+  -- service specifics
+  -- see https://github.com/aws/aws-sdk-js/blob/9295e45fdcda93b62f8c1819e924cdb4fb378199/lib/rds/signer.js#L11-L15
```

Comment:
Interesting...so the JS SDK consider this as part of the RDS "service class"

Reply:
exactly